src/app/pages/swap/swap-choose-asset/components/swap-asset-list.tsx

```diff
@@ -0,0 +1,40 @@
+import { useLocation, useNavigate } from 'react-router-dom';
+
+import { Box } from '@stacks/ui';
+import { useFormikContext } from 'formik';
+import get from 'lodash.get';
+
+import { SwapAsset, SwapFormValues } from '../../hooks/use-swap';
+import { SwapAssetItem } from './swap-asset-item';
+import { SwapAssetListLayout } from './swap-asset-list.layout';
+
+interface SwapAssetList {
+  assets: SwapAsset[];
+}
+export function SwapAssetList({ assets }: SwapAssetList) {
+  const { setFieldValue } = useFormikContext<SwapFormValues>();
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  async function onChooseAsset(asset: SwapAsset) {
```

Comment:
Maybe we can use `useLocationState` here?

Reply:
I hadn't seen that created, yep, I can replace it.